### PR TITLE
Fix parent/component mutation requirement

### PR DESCRIPTION
**Pull request by KevinBatdorf**
- Adds a refresh attribute to each watched component so that a mutation will always fire when the data changes. 
- Adds a check whether the magic property already exists
- Fixes a scenario when both a nested component and an external component attempt to watch the same component

Previously the component being watch would need to be re-identified each mutation. However, this change will just make sure that the mutation isn't from the component doing the watching, and will fetch the _watched_ component without that re-identification.

Fixes https://github.com/alpine-collective/alpine-magic-helpers/issues/9

### CODE REVIEW

**Comment by KevinBatdorf** [comment]
Should we wait on #7 to be merged in before this?

File: src/component.js [hunk 1]
```diff
@@ -18,12 +22,9 @@ function registerComponentMagicMethod() {
...
             for (let i = 0; i < mutations.length; i++) {
-                const closestParentComponent = mutations[i].target.closest('[x-data]')
-                if ((closestParentComponent && !closestParentComponent.isSameNode(parentComponent))) continue
-                if (!closestParentComponent.__x) {
-                    throw 'Error locating $parent data'
-                }
-                $el.$parent = allowTwoWayCommunication(closestParentComponent.__x.getUnobservedData(), parentComponent)
+                const mutatedComponent = mutations[i].target.closest('[x-data]')
+                if ((mutatedComponent && !mutatedComponent.isSameNode(parentComponent))) continue
+                $el.$parent = allowTwoWayCommunication(parentComponent.__x.getUnobservedData(), parentComponent)
```
Comment (SimoTod) [inline src/component.js:27]: you can probably return earlier as soon as you find a valid mutation, we don't need to keep looping on the array

File: package-lock.json [hunk 2]
```diff
@@ -1,13624 +0,0 @@
-{
```
Comment (SimoTod) [inline package-lock.json:1]: Why did we delete the lock file? It's hard to keep our node_module folder consistent without one and github actiona won't run

**Review by SimoTod — COMMENTED** [review]
It looks sensible. A couple of minor questions, thanks

File: package-lock.json (hunk 2 above)
Comment (KevinBatdorf) [reply package-lock.json:1]: Jus an accident. Nice catch

File: src/component.js (hunk 1 above)
Comment (KevinBatdorf) [reply src/component.js:27]: Updated. Thanks

**Review by SimoTod — APPROVED** [review]
Thanks

**Review by SimoTod — APPROVED** [review]

**Pull request merged**